dask/dataframe/utils.py

```diff
@@ -734,3 +736,53 @@ def drop_by_shallow_copy(df, columns, errors="raise"):
 
 class AttributeNotImplementedError(NotImplementedError, AttributeError):
     """NotImplementedError and AttributeError"""
+
+
+def serial_frame_constructor(like):
+    """Return a serial DataFrame constructor
+
+    Parameters
+    ----------
+    like :
+        Any series-like or dataframe-like object.
+    """
+    if is_dask_collection(like):
+        try:
+            like = like._meta
+        except AttributeError:
+            like = like.meta
+    if is_dataframe_like(like):
+        return like._constructor
+    elif is_series_like(like):
+        return like._constructor_expanddim
+    elif is_index_like(like):
+        return like.to_frame()._constructor
+    elif is_arraylike(like):
+        return serial_constructor_from_array(like)
+    else:
+        raise TypeError(f"{type(like)} not supported by serial_frame_constructor")
+
+
+def serial_series_constructor(like):
+    """Return a serial Series constructor
+
+    Parameters
+    ----------
+    like :
+        Any series-like or dataframe-like object.
+    """
+    if is_dask_collection(like):
+        try:
+            like = like._meta
+        except AttributeError:
+            like = like.meta
+    if is_dataframe_like(like):
+        return like._constructor_sliced
+    elif is_series_like(like):
+        return like._constructor
+    elif is_index_like(like):
+        return like.to_frame()._constructor_sliced
+    elif is_arraylike(like):
+        return serial_constructor_from_array(like, series=True)
```

Comment:
Including conversion from arrays adds an extra layer of complexity. Why is this needed?

Reply:
>Why is this needed?

It is not **needed** for this PR. I added it as an illustration for what would be needed to avoid passing the `like_df` argument ([see your comment here](https://github.com/dask/dask/pull/9570#discussion_r994767598)).  With that said, I do think it could be usefult for dask.dataframe to have a mechanism for creating a "serial" dataframe from array-like data.

Reply:
**Update**: I removed array handling and the related dispatch from this PR